mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -89,12 +93,27 @@ impl S3CrtClient {
 
         let mut client_bootstrap = ClientBootstrap::new(&allocator, &bootstrap_options).unwrap();
 
-        let creds_options = CredentialsProviderChainDefaultOptions {
-            bootstrap: &mut client_bootstrap,
+        let mut client_config = ClientConfig::new();
+
+        let mut credentials_provider = match config.profile_name_override {
```

Comment:
Should this be moved under the if at line 113?

Reply:
It should not, we need to bump ref count of `CredentialProvider` to 1 all the way until `S3CrtClient` owns the client config.

Reply:
I think I see what you mean, but maybe it's the signature of `init_default_signing_config` that's an issue: why is it taking a (mutable) reference instead of taking ownership of `CredentialProvider`? Is this a similar issue to what @jamesbornholt pointed out about the options parameter below? (or am I missing something?)